### PR TITLE
Add the new authentication header in Cloudflare v4

### PR DESCRIPTION
**Pull request by martinstreicher**
This PR has two things: 
* A bunch of Rubocop changes 
* A new authentication header in lib/cloudflair/connection.rb, which uses a token to replace the email address and key.  

I am also able to use this code to update a DNS record, which is reported as an Issue in this repo.

### CODE REVIEW

File: lib/cloudflair/connection.rb [hunk 1]
```diff
@@ -7,22 +9,22 @@ module Cloudflair
...
-      headers = {}
-      cloudflare_auth_config = Cloudflair.config.cloudflare.auth
-      if !(cloudflare_auth_config.key.nil? || cloudflare_auth_config.email.nil?)
-        headers['X-Auth-Key'] = cloudflare_auth_config.key
-        headers['X-Auth-Email'] = cloudflare_auth_config.email
-      elsif !cloudflare_auth_config.user_service_key.nil?
-        headers['X-Auth-User-Service-Key'] = cloudflare_auth_config.user_service_key
-      else
-        raise CloudflairError, 'Neither email & key nor user_service_key have been defined.'
+      {}.tap do |request_headers|
```
Comment (thde) [inline lib/cloudflair/connection.rb]: I'd probably prefer something like this:
```ruby
cloudflare_auth_config = Cloudflair.config.cloudflare.auth

unless cloudflare_auth_config.key.nil? || cloudflare_auth_config.email.nil?
  return { 'X-Auth-Key' => cloudflare_auth_config.key, 'X-Auth-Email' => cloudflare_auth_config.email }
end

unless cloudflare_auth_config.user_service_key.nil?
  return { 'Authorization' => "Bearer #{cloudflare_auth_config.user_service_key}" }
end

raise CloudflairError, 'Neither email & key nor user_service_key have been defined.'
```

Or `if !` if you're not a fan of `unless`.

File: .rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+require:
```
Comment (thde) [inline .rubocop.yml:1]: I'm not against the reformatting but maybe it should be checked in the Travis config aswell? @neckhair what do you think?

**Review by thde — CHANGES_REQUESTED** [review]
Thanks very much for your contribution! A small suggestion from my side.

File: lib/cloudflair/connection.rb (hunk 1 above)
Comment (martinstreicher) [reply lib/cloudflair/connection.rb]: That's a fine suggestion. I think I wanted to keep the structure of what was there, but I would be happier to make it look like this.

Comment (thde) [reply lib/cloudflair/connection.rb]: The `user_service_key` should probably also be reflected in the `README.md`?

Comment (martinstreicher) [reply lib/cloudflair/connection.rb]: Done.

Comment (martinstreicher) [reply lib/cloudflair/connection.rb]: > The `user_service_key` should probably also be reflected in the `README.md`?

The Configuration section mentions it? Do we need something else?

Comment (thde) [reply lib/cloudflair/connection.rb]: Nope looks perfect!

**Review by thde — APPROVED** [review]
Looks good 👍  Thanks very much for the effort!

**Comment by martinstreicher** [comment]
> Looks good 👍 Thanks very much for the effort!

Thank you. Someone will have to merge this for me since I don't have  a stinkin' badge.

**Comment by thde** [comment]
Jep let's wait on @neckhair's feedback on this.

**Review by neckhair — APPROVED** [review]
Thanks for your PR @martinstreicher! 
I like the inclusion of Rubocop here, but it would have been better to put it into a separate pull request. It's a bit hard to review the essentials here. Nevertheless I'll merge and release later.

**Pull request merged**